src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupCodeHelpers.cs

```diff
@@ -49,25 +51,75 @@ internal static unsafe void InitializeCommandLineArgs(int argc, byte** argv)
             Environment.SetCommandLineArgs(args);
         }
 
-        private static unsafe void InitializeStringTable()
+        /// <summary>
+        /// Each managed module linked into the final binary may have its own global tables for strings,
+        /// statics, etc that need initializing. InitializeGlobalTables walks through the modules
+        /// and offers each a chance to initialize its global tables.
+        /// </summary>
+        private static unsafe void InitializeGlobalTables()
         {
-            int length = 0;
-            IntPtr strEETypePtr = GetModuleSection((int)ModuleSectionIds.StringEETypePtr, out length);
-            Contract.Assert(length == IntPtr.Size);
-
-            IntPtr strTableStart = GetModuleSection((int)ModuleSectionIds.StringFixupStart, out length);
-            Contract.Assert(length % IntPtr.Size == 0);
-
-            IntPtr strTableEnd = (IntPtr)((byte*)strTableStart + length);
+            IntPtr* modulesSectionStart;
+            IntPtr* modulesSectionEnd;
+            GetModulesSection(out modulesSectionStart, out modulesSectionEnd);
+            
+            for (IntPtr* current = modulesSectionStart; current < modulesSectionEnd; current++)
```

Comment:
It may be nice to move this big loop to Initialize, and make this function work on a given module only.